master/internal/api_tasks.go

```diff
@@ -373,6 +375,66 @@ func (a *apiServer) TaskLogs(
 	})
 }
 
+func (a *apiServer) Monitor(ctx context.Context, taskID string, logs []*model.TaskLog) error {
+	// TODO(ft) do all logs have same taskID? Or should we just assume that they don't need to.
+	// TODO(ft) do all logs stream through here? K8S I think we are planning to add log through
+	// k8s api too.
+
+	isExp, exp, err := expFromTaskID(ctx, model.TaskID(taskID))
+	if err != nil {
+		return err
+	}
+	if isExp == false {
+		return nil
+	}
+
+	activeConfig, err := a.m.db.ActiveExperimentConfig(exp.ID)
+	if err != nil {
+		return err
+	}
+
+	for _, l := range logs {
+		if l.AgentID == nil {
+			return fmt.Errorf("agentID must be non nil") // TODO can we get away with this?
+			// It feels kinda annoying to let our database have a possible null that theoretically
+			// shouldn't happen.
+		}
+
+		for _, lpp := range activeConfig.LogPatternPolicies() {
+			regex := fmt.Sprintf("(.*)%s(.*)", lpp.Pattern())
+			r, _ := regexp.Compile(regex)
+
+			if r.MatchString(l.Log) {
+				policy := lpp.Policy().GetUnionMember()
+				switch reflect.TypeOf(policy).String() {
```

Comment:
use a type switch instead of this reflect, like here

https://github.com/determined-ai/determined/blob/8d2b5ce5b2134513a3c559d5545ad3e8c4fb3230/master/pkg/checkpoints/checkpoints.go#L40